### PR TITLE
Archive Rally logs in rally-tracks-compat job

### PR DESCRIPTION
**Pull request by pquentin**
This was needed to investigate #1688.

### CODE REVIEW

**Review by dliappis — APPROVED** [review]
LGTM thanks!

**Comment by dliappis** [comment]
@pquentin Where can we find the uploaded logs from GH actions? Some sort of pointer would be beneficial (even as a comment there)

**Comment by pquentin** [comment]
The artifact shows up under "Artifacts" in the "Summary" page: https://github.com/elastic/rally/actions/runs/4543523055?pr=1696 (which is counterintuitive, I would have expected to see it in the build itself)

image

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -67,6 +67,12 @@ jobs:
           # elastic/endpoint fetches assets from GitHub, authenticate to avoid
           # being rate limited
           ASSETS_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - uses: actions/upload-artifact@v3
```
[the remark this answers is not in the document]
Comment (pquentin) [reply .github/workflows/ci.yml:71]: Done in [`26fb761` (#1696)](https://github.com/elastic/rally/pull/1696/commits/26fb761c6a291074261864ddf0f2f41fbdf8b945)

**Review by danielmitterdorfer — APPROVED** [review]
Very useful. Thanks!

**Pull request merged**